test/fluree/db/shacl/shacl_basic_test.clj

```diff
@@ -1383,6 +1383,52 @@
       (is (util/exception? invalid-kid))
       (is (= "SHACL PropertyShape exception - path [[1002 :predicate]] conformed to sh:qualifiedValueShape fewer than sh:qualifiedMinCount times."
              (ex-message invalid-kid)))))
+  #_(testing "sh:qualifiedValueShape node shape"
```

Comment:
Is there a test that passes with these changes that wouldn't without them?

I know we have discussed this limitation
> we don't support having data that doesn't conform to a node shape that targets it.

and it's a tricky one. Will this be addressed in a followup PR?

Reply:
There is no existing test that passes now that wouldn't without these changes.

I don't have any plans for a followup PR, at least until we discuss the use case and iron out what the semantics should be.

Reply:
Since we talked about this more outside of Github, I'd like to capture some notes here so they can be more easily found later:
- Before this PR, if you used a node shape as a `sh:qualifiedValueShape`, you would get an internal error. Now, you would get a validation error if there's a violation of that shape.
- However, our current semantics for SHACL shapes is that any `sh:NodeShapes` that have been transacted will be enforced. So you might have a `sh:qualifiedValueShape` which permits some non-conforming data, and with this PR we would handle that correctly. But, you could still fail on the original `sh:NodeShape` , because we check all existing shapes
- How we handle this sort of conflict requires some more discussion/decision-making. The SHACL spec itself is about using shapes to generate validation reports, so it doesn't really specify semantics for this kind of issue. If you were just generating validation reports, you could generate one for either the `sh:qualifiedValueShape` shape or the `sh:NodeShape` it targets if you wanted, there's no conflict there. The idea of accumulating shapes and requiring that all transacted data conform is kind of our thing, so we need to figure out what that means in this case.
- Possible options (for the future) include:
   - support for active/inactive shapes
   - contexts in which a shape is inactive
   - inventing some kind of hierarchy/supersedence semantics

Does all that sound correct?

Reply:
I think you've distilled it very clearly and correctly : )

In plain 'ol SHACL you define the "shape graph" that you want to run against your "data graph". Well, our "shape graph" is all the `sh:NodeShape`s that we know, and so a user has no control over when those shapes get used for validation.

As for possible future implementations to support user control over the "shape graph", I'd like to see us carefully consider when and how users can express "optional" shapes, preferably with some feedback from actual users.

Reply:
I spoke about the concept of specifying one or more "control graphs" for a ledger - this would mean you could create a separate SHACL ledger, and then reuse those SHACL rules across multiple data ledgers without having to duplicate the rules, as you have to do today. This could allow separate teams, if desired, to manage global rules that are then enforced on other team's datasets.

If we have SHACL not specified as a "control graph", we don't enforce it - it is just plain data. So the way to get the identical behavior we have today would be to have the "control graph" be itself... and maybe that is a sensible default.

The idea is a control graph would be anywhere we perform enforcement or inferencing - so it isn't just SHACL, but also Fluree Policy rules, OWL, RDFS/generic vocabularies. They could all get reused,